CommonTools/Utils/interface/KinematicTables.h

```diff
@@ -27,6 +28,72 @@ namespace edm {
+                                       col::PtError,
+                                       col::MissingInnerHits,
+                                       col::NumberOfValidHits,
+                                       col::Charge,
+                                       col::Eta,
+                                       col::Phi,
+                                       col::Pt,
+                                       col::Theta,
+                                       col::D0>;
+    using TrackTableView = edm::soa::TableView<col::Px,
+                                               col::Py,
+                                               col::Pz,
+                                               col::P,
+                                               col::PtError,
+                                               col::MissingInnerHits,
+                                               col::NumberOfValidHits,
+                                               col::Charge,
+                                               col::Eta,
+                                               col::Phi,
+                                               col::Pt,
+                                               col::Theta,
+                                               col::D0>;
+    using TrackRowView = edm::soa::RowView<col::Px,
+                                           col::Py,
+                                           col::Pz,
+                                           col::P,
+                                           col::PtError,
+                                           col::MissingInnerHits,
+                                           col::NumberOfValidHits,
+                                           col::Charge,
+                                           col::Eta,
+                                           col::Phi,
+                                           col::Pt,
+                                           col::Theta,
+                                           col::D0>;
+
+    template <class Object>
+    TrackTable makeTrackTable(std::vector<Object> const& objects) {
+      return {objects,
+              edm::soa::column_fillers(
+                  col::D0::filler([](Object const& x) { return x.d0(); }),
+                  col::Px::filler([](Object const& x) { return x.px(); }),
+                  col::Py::filler([](Object const& x) { return x.py(); }),
+                  col::Pz::filler([](Object const& x) { return x.pz(); }),
+                  col::P::filler([](Object const& x) { return x.p(); }),
+                  col::Pt::filler([](Object const& x) { return x.pt(); }),
+                  col::Theta::filler([](Object const& x) { return x.theta(); }),
+                  col::PtError::filler([](Object const& x) { return x.ptError(); }),
+                  col::MissingInnerHits::filler([](Object const& x) {
+                    return x.hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS);
```

Comment:
This introduces a dependency of `CommonTools/Utils` on `DataFormats/TrackReco`. I'm not sure if these track-specific columns should still be defined in `CommonTools/Utils`. For sure this has to be somehow moved out of `KinematicColumns.h`, but I don't know where.

Reply:
I think organizing the column definitions in the `DataFormats` would be a rather natural way to go (but I'd be curious of hearing any better suggestions).

Reply:
Yes, I'm also curious. `DataFormats` sounds good in principle, but isn't it meant for persistent data formats? I remind that these SoA columns are only used within individual EGamma producers right now.

My personal opinion is that we are trying to centralize these SOA typenames prematurely. I'm leaning towards putting these track table and track specific column definitions into the `RecoEgamma/EgammaElectronAlgos` library itself and only move them out if there actually are common use cases that are not only limited to Egamma. @slava77 might see that different I think.